<a href="https://colab.research.google.com/github/harshitha-reddy10/flyrank-ml-internship-harshitha/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/harshitha-reddy10/flyrank-ml-internship-harshitha/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

Lane: Content performance prediction. I want to investigate which content and search-performance factors are associated with declining content, so that pages needing updates can be prioritized. I chose this lane because identifying declining pages early can help focus content-refresh efforts where they may have the greatest value.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Decision: Which declining content pages should be prioritized for refresh. The content or SEO team would use the recommendation to decide where to spend time and resources. A wrong recommendation could waste effort on a page that does not need updating while a genuinely declining, high-value page is overlooked.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
import pandas as pd

import os
import pandas as pd

if not os.path.exists("flyrank-ml-internship-harshitha"):
    !git clone https://github.com/harshitha-reddy10/flyrank-ml-internship-harshitha.git

df = pd.read_csv(
    "flyrank-ml-internship-harshitha/data/raw/content_refresh_anonymized.csv"
)

print("Total pages:", len(df))
print("Declining pages:", (df["trend_direction"] == "down").sum())
print(
    "Declining rate:",
    round((df["trend_direction"] == "down").mean() * 100, 1),
    "%"
)


Total pages: 30000
Declining pages: 16262
Declining rate: 54.2 %


The starter dataset contains 30,000 content pages. Of these, 16,262 pages are classified as declining, giving an observed declining rate of 54.2%. Since more than half of the pages show a downward trend, identifying which pages should be prioritized for refresh is a meaningful decision-support problem.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


This work can identify observed and directional patterns associated with declining content and provide decision support for prioritizing pages for refresh. The results may help the content or SEO team decide which pages deserve further review. However, the analysis cannot prove that any feature causes content decline, guarantee future performance, or claim to predict Google’s algorithm.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.